mobile/library/kotlin/io/envoyproxy/envoymobile/StreamCallbacks.kt

```diff
@@ -29,27 +30,46 @@ internal class StreamCallbacks {
  * `EnvoyHTTPCallbacks`.
  */
 internal class EnvoyHTTPCallbacksAdapter(
-  private val executor: Executor,
+  private val executor: Executor?,
   private val callbacks: StreamCallbacks
 ) : EnvoyHTTPCallbacks {
-  override fun getExecutor(): Executor {
-    return executor
-  }
-
   override fun onHeaders(
     headers: Map<String, List<String>>,
     endStream: Boolean,
     streamIntel: EnvoyStreamIntel
   ) {
-    callbacks.onHeaders?.invoke(ResponseHeaders(headers), endStream, StreamIntel(streamIntel))
+    if (executor != null) {
```

Comment:
so it looks like we're making the executor optional - what's the advantage of that?  Seems to put more onus on the caller.  What about having a default executor of DirectExecutor?

Reply:
If we are defaulting it to the direct executor, we always have to make a copy since there's no way we can omit the copy or else we'll risk having a dangling `ByteBuffer`. For Cronvoy, this would result in two copies (one from here and the other one on the Cronvoy side) and this is what I want to avoid, basically to eliminate the copy when it's not necessary, so making it optional is a better design IMO.

Another side note, the C++ API doesn't actually enforce some kind of threading model. It's basically up to the callback implementer to decide whether they need some kind of threading. Although the onus is on the caller, it provides more flexibility IMO.